polkadot/consensus/src/evaluation.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2017 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Polkadot block evaluation and evaluation errors.
+
+use super::MAX_TRANSACTIONS_SIZE;
+
+use codec::Slicable;
+use polkadot_runtime::Block as PolkadotGenericBlock;
+use polkadot_primitives::Timestamp;
+use polkadot_primitives::parachain::Id as ParaId;
+use primitives::block::{Block as SubstrateBlock, HeaderHash, Number as BlockNumber};
```

Comment:
ugh. this is a step backwards - my PR removes this datastructure entirely and there's already an upgrade path. rather than it being a substrate block, just make it `B`, `where B: substrate_runtime_primitives::traits::Block`.

Reply:
it's not yet usable without significant changes spanning all of substrate (network, client, bft in particular). I'd classify that as beyond the scope of this PR and IMO it's not really a blocker -- once those modules are made generic, this code will follow easily.